Add pieces sti

zenkbaries opened the pull request:
Added Single Table Inheritance for Pieces for the following pieces:
Pawn
Knight
Bishop
Rook
Queen
King

and added scopes to Pieces Models for each pieces

zenkbaries committed:
Add STI table

zenkbaries committed:
Added scopes to paren model

zenkbaries committed:
Clean up STI models

zenkbaries committed:
clean up piece.rb

mscoutermarsh commented on app/models/piece.rb:
```diff
@@ -1,4 +1,11 @@
+  belongs_to :game
 end
+
+scope :pawns, -> { where(race: 'Pawn') }
```
Scopes would have to be within the class (see line #4). Above the end

mscoutermarsh commented:
Almost done!

To finish this, you'll need to add a `type` column to `Piece`.

In the guide I sent you, they use `race`. Which I'm now realizing is confusing. Ignore that and use `type` instead.

`type` is the default that Rails uses for single table inheritance.

mscoutermarsh commented on app/models/piece.rb:
```diff
@@ -1,4 +1,11 @@
+scope :bishops, -> { where(race: 'Bishop') }
+scope :rooks, -> { where(race: 'Rook') }
+scope :queens, -> { where(race: 'Queen') }
+scope :kings, -> { where(race: 'King') }
```
instead of `race` let's use `type`

zenkbaries committed:
Moved the 'scope' line to inside class/end block

zenkbaries commented:
Ok I replaced all 'race' with 'type'
and the type column is already in pieces from last week migration file.

I had a conflict and resolved (seems that pieces from master file is different from add-pieces-sti file which i know I change but they call it 'conflict'  i dont know why) so i fixed it and now it merge successfully  (not realized that I should have asked @mscoutermarsh for review before 'merge' it)
